### ISSUE TITLE
soft lockup cpu on galaxy book s

### ISSUE THREAD

**Issue by nullexistence** [aarch64-laptops/debian-cdimage#16]
![DSC_0060](https://user-images.githubusercontent.com/85072957/120112430-677f6200-c1a8-11eb-98a8-b90dae62cedf.JPG)

**Comment by TheOneWithTheBraid** [comment]
Could you solve this? I got similar problems too, even though I tried newer kernels too (which should theoretically support the Snapdragon 8cx).

**Comment by ghost** [comment]
I found a workaround, it will sound absolutely insane but still.
I just plugged a second usb stick into the laptop.

My setup is:
Left usb port with installer
Right port with some random fat32 stick.
Have sadly no time to tinker atm and have done no more testing just saw that I was suddenly sitting in the Debian installer.


On a unrelated note. Does any one know what this image does differently to any other arm64 installer from other distros? Fedora and any other just reboot after grub.
![20220611_161135_1](https://user-images.githubusercontent.com/67808020/173280921-dfbc5448-d350-4df2-8058-ce46f7437890.jpg)

**Comment by ghost** [comment]
Finally installed it, and got it to boot I to multiuser target

**Comment by ghost** [comment]
![20220612_100201_1](https://user-images.githubusercontent.com/67808020/173280995-c69fdc58-414a-4368-ab29-d6deba22f0fe.jpg)

Charger is not working

**Comment by TheOneWithTheBraid** [comment]
For me, this didn't work. Could you maybe add some more words on what you did before?

In my case, the Grub simply reboots - even when having the second USB drive connected.

Did you use the Debian image linked in the release https://github.com/aarch64-laptops/debian-cdimage/releases/tag/v0.4?

What device tree blob did you load? Did you change any settings in the UEFI settings?

**Comment by ghost** [comment]
Yes, I used debian-unstable-arm64-DVD-1_20210314.iso 
with no changes to device tree. While installing I used two usbc -> usba adapters with usb sticks. But later realized that one usbc -> usba adapter and my usbc -> ethernet adapter also did the trick.
Check the first photo by nullexistance. I get the cpu lockup from his photo everytime I have just one port occupied.

Edit:
I think it's a problem with power delivery and having those usb2.0 devices maybe somehow works around some problem.

**Comment by TheOneWithTheBraid** [comment]
Ah, I found my problem: Using a USB 3 drive simply didn't work. Switching to a USB 2 drive did the trick.

**Comment by misyltoad** [comment]
How did you manage to use `debian-unstable-arm64-DVD-1_20210314.iso`?

I dd'ed it to a flash drive and my boot menu doesn't pick up on it. I tried dd'ing a raw (not ISO) Fedora image and it detects it, but as noted above doesn't work.

Seems like my Book S doesn't detect any isohybrid images.

**Comment by misyltoad** [comment]
I committed some horrible sins and dumped the ISO contents all onto a vfat partition, and that at least got me to the soft lockup

**Comment by misyltoad** [comment]
The double USB trick worked here too, haha amazing!

**Comment by misyltoad** [comment]
I got the error "Failed to determine the codename for the release" but worked around by plugging in an ethernet cable to my Dock.

**Comment by pierro78** [comment]
thanks for sharing the double usb trick !!

here are some newbie notes/questions : 

apparently I still need the double usb trick to boot from internal ssd ...
... maybe this could be fixed by installing a newer kernel as told in https://github.com/aarch64-laptops/debian-cdimage ?? but which version ?? is **5.19** available ??

also gdm3 was giving me a black screen but I managed to open an alternative console (with "Ctrl Alt F2") and I installed the openssh-server to work more easily from another PC ... but I needed an ethernet cable to connect to the galaxy book S ... anybody has found how to connect the galaxy book S through **wifi** ?? 
... then I installed xfce and using lightdm worked to  start my DEs !! (gnome was kind of slow but xfce feels faster ...)

also as @systemscape said the charger doesn't seem to charge ... anybody found a solution to get the galaxy book S to **charge** ???
... and what tool can we use to check if the galaxy book S is actually charging ??


also from https://github.com/aarch64-laptops/debian-cdimage **"Firmware installation"** it looks like we need to load some kind of firmware for the galaxy book S to get some hardware blocks working ?? but how do we do that ??


thanks folks for any help !!

... oops ... **sound** doesn t seem to work ... did anybody find a way to make it work ??
... thanks again !

PS1 : installed the linux arm64 thorium browser (with google sync) from https://github.com/Alex313031/Thorium-Special/releases ... only 6725 in octane v2 (got about 32500 with Edge on Windows ...) ... chromium only 6493 ... firefox feels slow too ... anybody has an idea how to **improve performance** ??

PS2 2022 08 11 : here is an answer I got on https://webchat.oftc.net/?channels=%23aarch64-laptops (archive : https://oftc.irclog.whitequark.org/aarch64-laptops/2022-08-11 ) (not sure I understand everything) **(see also comment below)** :  

[15:39](https://oftc.irclog.whitequark.org/aarch64-laptops/2022-08-11#31223646) <jenneron[m]> pierro78: i started to work on samsung galaxy book s, but it needs multi port usb for keyboard/touchpad, there are 5.19-based branches with support for it, but on 5.19 pmic arbiter fails..
[15:41](https://oftc.irclog.whitequark.org/aarch64-laptops/2022-08-11#31223648) <jenneron[m]> if you want to play around with it e.g. to bisect some kernel versions, you can use my dtb https://gitlab.com/jenneron/linux/-/commit/76f402e89cfb8308eb44822eda8fc816282d74a1, the latest version which works was 5.15 rc something

**Comment by misyltoad** [comment]
Something broke for me and now I always get dumped straight into grub minimal on the same flash drive I had working before argh

**Comment by pierro78** [comment]
here is an answer I got on https://webchat.oftc.net/?channels=%23aarch64-laptops (archive : https://oftc.irclog.whitequark.org/aarch64-laptops/2022-08-11 ) (not sure I understand everything) :

[15:39](https://oftc.irclog.whitequark.org/aarch64-laptops/2022-08-11#31223646) <jenneron[m]> pierro78: i started to work on samsung galaxy book s, but it needs multi port usb for keyboard/touchpad, there are 5.19-based branches with support for it, but on 5.19 pmic arbiter fails..
[15:41](https://oftc.irclog.whitequark.org/aarch64-laptops/2022-08-11#31223648) <jenneron[m]> if you want to play around with it e.g. to bisect some kernel versions, you can use my dtb https://gitlab.com/jenneron/linux/-/commit/76f402e89cfb8308eb44822eda8fc816282d74a1, the latest version which works was 5.15 rc something

PS :  
device tree : 
https://gitlab.com/jenneron/linux/-/tree/galaxy-book-s

**Comment by pierro78** [comment]
progress as seen on https://oftc.irclog.whitequark.org/aarch64-laptops/2022-08-17 :  

[20:11](https://oftc.irclog.whitequark.org/aarch64-laptops/2022-08-17#31264981) <jenneron[m]> qzed: thank you, disabling pmic@a and pmic@2 does the trick
[20:11](https://oftc.irclog.whitequark.org/aarch64-laptops/2022-08-17#31264982) <jenneron[m]> also, your usb-mp patchset gives working keyboard and touchpad on samsung galaxy book s

**Comment by pierro78** [comment]
progress thanks to jenneron and the aarch64-laptops group : newer kernel 5.19.0 with patches, dtb, firmware ... **still working on it** to add stuff (like, for example, wifi) ... octane v2 28228 on thorium instead of 6725 with debian-cdimage kernel ... can boot with an empty usb slot ... all the "dirty details" are in the irc archive, for example https://oftc.irclog.whitequark.org/aarch64-laptops/2022-08-21 ...


just adding some notes here :  
. grub.cfg entry : add "devicetree /path/to/dtb" (somewhat similar to https://dpaste.com/C9AWZJTFA)
. I built the kernel without ufs support but it seems to work fine from internal ssd ...
. firmware files into  /lib/firmware/qcom/sc8180x/sm-w767 and /lib/firmware/qcom

**Comment by pierro78** [comment]
progress : wifi working (but no touchscreen, and no charging) (dirty details in https://oftc.irclog.whitequark.org/aarch64-laptops/2022-08-23 )

**Comment by IDavideI** [comment]
Hi everyone, i'm also trying to make linux working on the samsung galaxy book S 2020. Mine is with intel lakefield, and I noticed that Fedora 36 and Pop!_OS work on the computer (touchpad, touchscreen and keyboard). Pop!_OS is better because Fedora 36 crashes a lot of time. 2 things do not work:
- the soundcard is not detected, thus there is no audio and no microfone;
- the battery isn't recognized and you can't see how much battery is left. Anyway if the computer is connected to the power it get in charging. 
I don't know if these informations are useful. If you have some advice for me, then share with me please. Thanks

**Comment by monkeyfart34** [comment]
> How did you manage to use `debian-unstable-arm64-DVD-1_20210314.iso`?
> 
> I dd'ed it to a flash drive and my boot menu doesn't pick up on it. I tried dd'ing a raw (not ISO) Fedora image and it detects it, but as noted above doesn't work.
> 
> Seems like my Book S doesn't detect any isohybrid images.

Same here, I'm able to boot to grub but its just restarts into windows past that

**Comment by pierro78** [comment]
> Same here, I'm able to boot to grub but its just restarts into windows past that

you need to add DtbLoader.efi as explained in https://github.com/aarch64-laptops/debian-cdimage
also you need 2 USB2 sticks for the system to run with this debian-unstable-arm64-DVD-1_20210314.iso ...


I could build an iso with jenneron's kernel, dtb and firmware but I don t know how to do it ... need to find some instructions ...

PS : here are some instructions to build a debian iso installer : https://github.com/Mani-Sadhasivam/debian-cdimage/tree/main/doc ( as seen on https://oftc.irclog.whitequark.org/aarch64-laptops/2022-09-01 )

**Comment by pierro78** [comment]
> I could build an iso with jenneron's kernel, dtb and firmware but I don t know how to do it ... need to find some instructions ...

in the meantime I am "releasing" the packages made from jenneron's kernel v5.19 sources here : https://github.com/pierro78/galaxybooks8cxkerneldeb/releases/tag/v0.1

i am going to explain how to install these below

**Comment by pierro78** [comment]
> i am going to explain how to install these below

these kernel v5.19 packages and files bring the following avantages over the 5.11custom kernel that is included in the debian-unstable-arm64-DVD-1_20210314.iso :  
. no need to keep usb sticks in usb ports
. speed (octane v2 about 28500 instead of 6500)
. wifi

however it is still very much a work in progress :
. touchscreen not working
. not drawing power from charger ? (jenneron released another dtb where the laptop would draw some power (like 6W) but not enough to charge) (actually no : I tried again and it now charges for a minute at the beginning (like 30W) and then it only draws like 5W)
. seems like external usb storage is not working ? (maybe jenneron released some other dtbs where this was working) (strange thing is I can boot debian linux on external usb, but then it sees only the external usb storage with debian as /dev/sda and not the others I plugin ...) (oh : when I plugin my usb storage in my usb c hub then it sees it !)
. no sound
. minimal kernel (some features are not activated ... like, for example, network bridge needed by docker)

**Installation :** 
. install the .deb packages
. copy the dtb to /boot and add a line like "devicetree /boot/dtbfilename" to your grub entry, update grub
. untar the firmware file from / directory (it will untar the contained files in /lib/firmware/qcom ... )

I hope this will bring some attention to the remarkable work that jenneron has done with the help of the aarch64-laptops group and that we'll have a better functionning linux on our snapdragon galaxy book S in the future ...

**2022 09 14 CAUTION NOTE**

@Joshua-Ashton tried to use these files/packages but it didn t work for him. I don't know why (I ve got debian installed with these files/packages (I think) on 2 partitions and these 2 installations work)

but according to https://oftc.irclog.whitequark.org/aarch64-laptops/2022-09-13 , if I get it right, @Joshua-Ashton was able to get a working 5.19 kernel on his galaxy book s ...

**Comment by ghost** [comment]
I'm grateful that there is progress on this laptop.
Thanks especially to jenneron for his work and time he contributed into pushing this forward.

@pierro78 you should add a word of caution for using those packages you assembled. Like you said this is testing quality. Jenneron device is also starting to suffer permanent display glitches.
Mine looks unaffected though.

**Comment by misyltoad** [comment]
> you need to add DtbLoader.efi as explained in https://github.com/aarch64-laptops/debian-cdimage

I didn't get that automatically placed for me when I used the installer :/

**Comment by misyltoad** [comment]
I copied the DTBLoader from the deb and it still doesn't boot... Maybe its missing the dtb files or something idk.

**Comment by pierro78** [comment]
> you need to add DtbLoader.efi as explained in https://github.com/aarch64-laptops/debian-cdimage 

@Joshua-Ashton  : you need to follow the instructions in the "AArch64 quirks" section of https://github.com/aarch64-laptops/debian-cdimage  to get the "AArch64 EFI" to load/boot linux ... this implies starting Shell.efi to run a bcfg command to modify the linux EFI boot entry ...

**Comment by misyltoad** [comment]
No what I mean is there was no dtbloader for the bcfg to use. It was not present after install. I will try re-installing today and maybe it'll work.

**Comment by misyltoad** [comment]
Okay I reinstalled it and got it to kinda boot. It gets onto kernel cmdline, I keep the two sticks in so it doesn't lockup and it hangs on a black screen with a `_` after that.

**Comment by misyltoad** [comment]
It seems stuff goes to crap when it starts graphical session for me... I can get a bit of tty by spamming `ctrl + alt + FX` before boot, but after gdm starts or whatever it goes to crap and freezes :S

**Comment by misyltoad** [comment]
Managed to use `text` on cmdline to get in and disable gdm + enable ssh

**Comment by pierro78** [comment]
> Managed to use `text` on cmdline to get in and disable gdm + enable ssh

@Joshua-Ashton does "gnome-shell --wayland" (or sthg like that) give you a graphical desktop ??

**Comment by misyltoad** [comment]
No `Failed to create backend: No drm devices found`

**Comment by misyltoad** [comment]
The 5.19 kernel doesn't work for me at all, just entirely black screens. I even tried building it + the device tree myself and it still didn't work. So I have no idea what's going on.

I get no fbcon output at all, I think it's failing very early on. Not sure how I can get more details.

**Comment by pierro78** [comment]
strange @Joshua-Ashton ... I guess you did an "apt update ; apt dist-upgrade" and un-tarred the firmware stuff ? does wifi work for you ?

**Comment by misyltoad** [comment]
Wifi doesn't work, but that's on 5.11, I can't boot into 5.19, it doesn't work.

**Comment by misyltoad** [comment]
I got it working a little bit ago.

You'll probably want my Mesa PR here to fix some rendering issues.
https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/18779

**Comment by monkeyfart34** [comment]
how do i re isntall windows?

**Comment by monkeyfart34** [comment]
Has anyone got wifi working? I got debian installed but I need wifi as school starts on monday

**Comment by ghost** [comment]
@monkeyfart34 
> Has anyone got wifi working? I got debian installed but I need wifi as school starts on monday

Hey, there is probably a way to get wifi working. But for school i definitely recommend you using windows.
Do need help with reinstalling windows? I have done that a bunch of times.

**Comment by pierro78** [comment]
> Has anyone got wifi working? I got debian installed but I need wifi as school starts on monday

yeah I got wifi working (see https://github.com/aarch64-laptops/debian-cdimage/issues/16#issuecomment-1231888011 ) but there are still important issues (see also https://github.com/aarch64-laptops/debian-cdimage/issues/16#issuecomment-1231888011 ) so I do not recommend linux for school yet ...

**Comment by monkeyfart34** [comment]
> @monkeyfart34
> 
> > Has anyone got wifi working? I got debian installed but I need wifi as school starts on monday
> 
> Hey, there is probably a way to get wifi working. But for school i definitely recommend you using windows. Do need help with reinstalling windows? I have done that a bunch of times.

Yeah, how would I reinstall windows? Right now without a usb i boot into BSOD 0xc00000e

**Comment by ghost** [comment]
> > @monkeyfart34
> > > Has anyone got wifi working? I got debian installed but I need wifi as school starts on monday
> > 
> > 
> > Hey, there is probably a way to get wifi working. But for school i definitely recommend you using windows. Do need help with reinstalling windows? I have done that a bunch of times.
> 
> Yeah, how would I reinstall windows? Right now without a usb i boot into BSOD 0xc00000e

You need to add the galaxy book drivers to the installation files on your usb.
I uploaded the drivers from my galaxy book to a telegram channel https://t.me/systemscape
I also wrote down what i documented when i did it.
The most important thing is, that you have the zip with the drivers. There are also tutorials on the web how to add drivers to a windows stick.
Ill probably push it also as a github repo later.

Edit:
https://github.com/mordaut97/galaxy_book_s_driver_pack

**Comment by monkeyfart34** [comment]
> > > @monkeyfart34
> > > > Has anyone got wifi working? I got debian installed but I need wifi as school starts on monday
> > > 
> > > 
> > > Hey, there is probably a way to get wifi working. But for school i definitely recommend you using windows. Do need help with reinstalling windows? I have done that a bunch of times.
> > 
> > 
> > Yeah, how would I reinstall windows? Right now without a usb i boot into BSOD 0xc00000e
> 
> You need to add the galaxy book drivers to the installation files on your usb. I uploaded the drivers from my galaxy book to a telegram channel https://t.me/systemscape I also wrote down what i documented when i did it. The most important thing is, that you have the zip with the drivers. There are also tutorials on the web how to add drivers to a windows stick. Ill probably push it also as a github repo later.

Okay, I got windows installed, but I don't have an option for Wi-Fi, not in device manager or anything, only Ethernet, Anyway I can fix this?

**Comment by ghost** [comment]
> > > > @monkeyfart34
> > > > > Has anyone got wifi working? I got debian installed but I need wifi as school starts on monday
> > > > 
> > > > 
> > > > Hey, there is probably a way to get wifi working. But for school i definitely recommend you using windows. Do need help with reinstalling windows? I have done that a bunch of times.
> > > 
> > > 
> > > Yeah, how would I reinstall windows? Right now without a usb i boot into BSOD 0xc00000e
> > 
> > 
> > You need to add the galaxy book drivers to the installation files on your usb. I uploaded the drivers from my galaxy book to a telegram channel https://t.me/systemscape I also wrote down what i documented when i did it. The most important thing is, that you have the zip with the drivers. There are also tutorials on the web how to add drivers to a windows stick. Ill probably push it also as a github repo later.
> 
> Okay, I got windows installed, but I don't have an option for Wi-Fi, not in device manager or anything, only Ethernet, Anyway I can fix this?

Sorry, have not had that one. I think your best bet would be to connect your phone with usb tethering or use a usb ethernet adapter to let windows update fix this.

**Comment by monkeyfart34** [comment]
> > > > > @monkeyfart34
> > > > > > Has anyone got wifi working? I got debian installed but I need wifi as school starts on monday
> > > > > 
> > > > > 
> > > > > Hey, there is probably a way to get wifi working. But for school i definitely recommend you using windows. Do need help with reinstalling windows? I have done that a bunch of times.
> > > > 
> > > > 
> > > > Yeah, how would I reinstall windows? Right now without a usb i boot into BSOD 0xc00000e
> > > 
> > > 
> > > You need to add the galaxy book drivers to the installation files on your usb. I uploaded the drivers from my galaxy book to a telegram channel https://t.me/systemscape I also wrote down what i documented when i did it. The most important thing is, that you have the zip with the drivers. There are also tutorials on the web how to add drivers to a windows stick. Ill probably push it also as a github repo later.
> > 
> > 
> > Okay, I got windows installed, but I don't have an option for Wi-Fi, not in device manager or anything, only Ethernet, Anyway I can fix this?
> 
> Sorry, have not had that one. I think your best bet would be to connect your phone with usb tethering or use a usb ethernet adapter to let windows update fix this.

OK, I'm rebuilding the ISO and I get this error

```
PS C:\> Add-WindowsDriver -Path "C:\mount\boot" -Driver "C:\drivers"
Add-WindowsDriver : An error occurred. No operation was performed.
Verify that DISM is installed properly in the image, and then try the operation again.
At line:1 char:1
+ Add-WindowsDriver -Path "C:\mount\boot" -Driver "C:\drivers"
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Add-WindowsDriver], COMException
    + FullyQualifiedErrorId : Microsoft.Dism.Commands.AddWindowsDriverCommand
```

**Comment by mothenjoyer69** [comment]
re: screen issues, has anyone else experienced the same failure as jenneron? on that note, is there any active work being done right now for this specific laptop? I know jenneron has put it on hold due to replacing their Galaxy Book S, but I haven't been able to find any other branches.

**Comment by mothenjoyer69** [comment]
https://github.com/mothenjoyer69/w767-linux

I squashed jennerons patches into a 6.x branch (Surface X branch, as it has a lot of SC8180x work) and got the reserved memory ranges working; in theory, battery and wifi should work fine together now (although i'm not actually sure how to enable the battery...)

edit: https://gitlab.com/jenneron/linux/-/tree/galaxy-book-s-6.0.6 have been talking to jenneron, and they've rebased on 6.0.6 :)

**Comment by pierro78** [comment]
> https://github.com/mothenjoyer69/w767-linux
> 
> I squashed jennerons patches into a 6.x branch (Surface X branch, as it has a lot of SC8180x work) and got the reserved memory ranges working; in theory, battery and wifi should work fine together now (although i'm not actually sure how to enable the battery...)
> 
> edit: https://gitlab.com/jenneron/linux/-/tree/galaxy-book-s-6.0.6 have been talking to jenneron, and they've rebased on 6.0.6 :)

awesome !! - did you have time to test ??? is charging and sound working now ???
(wifi was already working with jenneron's kernel I shared in https://github.com/aarch64-laptops/debian-cdimage/issues/16#issuecomment-1231898585 ...)

I am eager to test !! How can I do it ?? which kernel sources should I get ???

thanks !!!

**Comment by mothenjoyer69** [comment]
> 

unfortunately not yet! previously the issue was an issue with the reserved memory ranges, so now its actually possible to get charging and battery working. 

grab a copy of jennerons [galaxy-book-s-6.0.6-battery fork](https://gitlab.com/jenneron/linux/-/tree/galaxy-book-s-6.0.6-battery), build it, and boot it. i'm typing this reply from Debian Sid under Gnome 43 with Firefox ;)


(although, please be aware of jennerons screen issues in the past and the fact that we have no idea what caused them)

**Comment by pierro78** [comment]
thanks a lot !! no screen issues with linux here !! I am only aware of jenneron having screen issues with linux right now ...

**Comment by misyltoad** [comment]
Battery stuff doesn't work with 6.0.6-battery branch kernel with the DTS for me.

**Comment by matdomis** [comment]
Hey guys! Any updates? I'm really looking forward to use linux in this machine.

Is any linux distro working fine on Book S with the latest kernel updates??

I really need for my school/work.

Also do you recommend making a Dual Boot to not delete windows if any thing goes wrong with the linux OS?

**Comment by pierro78** [comment]
> Any updates? 

I am following the devs work on the aarch64-laptops irc and to my understanding there has not been notable progress lately although jenneron has worked on his Snapdragon Book S recently

**Comment by mothenjoyer69** [comment]
@pierro78 @matdomis Unfortunately nothing from my end; My Galaxy Book S has been loaned out for the foreseeable future, and I haven't quite been able to stomach the cost of buying a second unit yet. It is a device I am keen to return to when I am able to. It does look like jenneron has done some work lately, although I haven't touched base with them in a while.

**Comment by SNF444** [comment]
Hey all

Just wondering about the potential NAS applications with the work being done here. At this stage from my understanding is that the charging has issues but I'm wondering if enough power is delievered to keep the device on indefinitely as long as it remains plugged in ?

If thats the case does anyone know of any potential issues running things like docker or OMV, any virtual machines and shares etc.

It wouldn't require graphical desktops and the like, aswell touchscreen and similar features, acting only as a low powered NAS just for the sake of it.

**Comment by pierro78** [comment]
@SNF444 : the kernel provided in https://github.com/aarch64-laptops/debian-cdimage/issues/16#issuecomment-1231888011 is minimal and some features are not activated ... like, for example, network bridge needed by docker ... so you'd need to recompile your kernel if you want to do docker stuff with an up-to-date kernel ...

**Comment by SNF444** [comment]
Thanks for the quick reply , and alright I think thats doable I'll give it ago.

**Comment by pierro78** [comment]
comment from "agl7" on the aarch64-laptops irc : 
"As a Linux system I got **Armbian** to work. But not really either, there are always kernel panic errors."

https://oftc.irclog.whitequark.org/aarch64-laptops/2023-04-08
.

**Comment by SlowKnight0** [comment]
Hey I'm very much a novice when it comes to regular linux let alone talking about kernels and what not.

I've followed this page in a regular x64 Ubuntu vm:

https://github.com/Mani-Sadhasivam/debian-cdimage/tree/main/doc

And this error appeared in the very first of the Build kernel deb

<img width="355" alt="Screenshot 2023-04-09 130615" src="https://user-images.githubusercontent.com/130259601/230752025-dd656bab-8552-4fcf-8664-cb48a3b4bf22.png">

When I look for the lexer.lex.c it isn't in the respoitory if thats correct.

Can anyone point in the right direction ?

**Comment by pierro78** [comment]
> Hey I'm very much a novice when it comes to regular linux let alone talking about kernels and what not.
> 
> I've followed this page in a regular x64 Ubuntu vm:
> 
> https://github.com/Mani-Sadhasivam/debian-cdimage/tree/main/doc
> 
> And this error appeared in the very first of the Build kernel deb
> 
> <img alt="Screenshot 2023-04-09 130615" width="355" src="https://user-images.githubusercontent.com/130259601/230752025-dd656bab-8552-4fcf-8664-cb48a3b4bf22.png">
> 
> When I look for the lexer.lex.c it isn't in the respoitory if thats correct.
> 
> Can anyone point in the right direction ?

are you trying to cross compile your ARM Snapdragon Galaxy Book S kernel ?
it might be easier to compile it under windows on your ARM Snapdragon Galaxy Book S  with WSL (that s what I did)

you have requirements (see https://wiki.ubuntu.com/Kernel/BuildYourOwnKernel ) before compiling your kernel
"flex not found" means you have not installed the "flex" package (see for example https://wiki.ubuntu.com/Kernel/BuildYourOwnKernel )

**Comment by SlowKnight0** [comment]
Yes I think, I'm trying to build an installer for both your version and the battery fork, to install either alongside windows or standalone on the galaxy book s.

I did originally begin under wsl on the device itself but encountered the same issue (thanks lots for the answer)

I'm not entirely sure how to specifically target those source packages so I was going to start by following the guide I linked above.

Thanks for the help, and do you see any issues with my current path? 

I had thought that building those sources into an image alone might mean that I could skip most of the steps but I'm thinking they will all still need to be done.

**Comment by pierro78** [comment]
> I'm not entirely sure how to specifically target those source packages so I was going to start by following the guide I linked above.
> 
> Thanks for the help, and do you see any issues with my current path?

not really, myself I just followed the "Kernel Upgrade" section of https://github.com/aarch64-laptops/debian-cdimage and did not have any issue building my kernel packages from jenneron's sources ( aarch64-laptops irc ) 
 (note that in the "Kernel Upgrade" section of https://github.com/aarch64-laptops/debian-cdimage they tell you how to install flex, etc ...)

**Comment by SlowKnight0** [comment]
Ah this crossed my mind, perhaps I should just do this instead.

So instead of I might do this.

When following your instructions on copying the dtb and installing the .deb etc.

Do I intially boot from the latest iso under 

https://github.com/aarch64-laptops/debian-cdimage/releases

and follow those instructions with the your resource ? here

https://github.com/pierro78/galaxybooks8cxkerneldeb/releases/tag/v0.1

Reading those instructions it talks about firmware related to the flex 5g, guessing I just skip over those parts.

**Comment by oktaykarakiya** [comment]
I'd like to help out with the coding stuff, i dont have much experience with these things, but i can code, and if someone could help me a little initially i think that i could dedicate some time to code stuff for this wonderful device

**Comment by LeosDing** [comment]
All done under galaxy book s w767?

**Comment by pierro78** [comment]
> All done under galaxy book s w767?

sure, you are in the AArch64-laptops github project here ....

**Comment by pierro78** [comment]
> I'd like to help out with the coding stuff, i dont have much experience with these things, but i can code, and if someone could help me a little initially i think that i could dedicate some time to code stuff for this wonderful device

did you try to talk about your project in the aarch64-laptops irc ??

**Comment by LeosDing** [comment]
> pierro78 reacted with thumbs up emoji

I have encountered the same problem as you, and I also need a random FAT32 USB drive to start. Otherwise, there will still be lockup issues.

How did you solve it later?

**Comment by LeosDing** [comment]
> Okay I reinstalled it and got it to kinda boot. It gets onto kernel cmdline, I keep the two sticks in so it doesn't lockup and it hangs on a black screen with a `_` after that.

I have encountered the same problem as you, and I also need a random FAT32 USB drive to start. Otherwise, there will still be lockup issues.

How did you solve it later?

**Comment by pierro78** [comment]
> How did you solve it later?

I think he re-compiled the kernel ( I have also recompiled it and made it available, see https://github.com/aarch64-laptops/debian-cdimage/issues/16#issuecomment-1231888011 )

**Comment by LeosDing** [comment]
> > How did you solve it later?
> 
> I think he re-compiled the kernel ( I have also recompiled it and made it available, see [#16 (comment)](https://github.com/aarch64-laptops/debian-cdimage/issues/16#issuecomment-1231888011) )

In my case, the 5.11 kernel you compiled must have two typec HUBs plugged in to boot properly from the UFS disk.

**Comment by pierro78** [comment]
> In my case, the 5.11 kernel you compiled must have two typec HUBs plugged in to boot properly from the UFS disk.
 
 I compiled Jenneron's kernel sources and they are 5.19 and no need for USB sticks to boot, see https://github.com/aarch64-laptops/debian-cdimage/issues/16#issuecomment-1231888011 , see also the comment below for more details : https://github.com/aarch64-laptops/debian-cdimage/issues/16#issuecomment-1231898585

**Comment by LeosDing** [comment]
> > In my case, the 5.11 kernel you compiled must have two typec HUBs plugged in to boot properly from the UFS disk.
> 
> I compiled Jenneron's kernel sources and they are 5.19 and no need for USB sticks to boot, see [#16 (comment)](https://github.com/aarch64-laptops/debian-cdimage/issues/16#issuecomment-1231888011) , see also the comment below for more details : [#16 (comment)](https://github.com/aarch64-laptops/debian-cdimage/issues/16#issuecomment-1231898585)

Is your DTB compiled from the 5.19 kernel? 
Is this DTB file placed in /boot and added as a devicetree in grub.cfg? 
Or was it directly thrown into the /boot/efi/dtb directory?

**Comment by pierro78** [comment]
> Is your DTB compiled from the 5.19 kernel? Is this DTB file placed in /boot and added as a devicetree in grub.cfg?

Exactly ! ( as I said in https://github.com/aarch64-laptops/debian-cdimage/issues/16#issuecomment-1231898585 )

**Comment by LeosDing** [comment]
> > Is your DTB compiled from the 5.19 kernel? Is this DTB file placed in /boot and added as a devicetree in grub.cfg?
> 
> Exactly ! ( as I said in [#16 (comment)](https://github.com/aarch64-laptops/debian-cdimage/issues/16#issuecomment-1231898585) )

Following your steps, the 5.19 kernel you compiled will display the following boot time:
`
EFI stub: Booting Linux Kernel...
EFI Stub: Using DTB from configuration table
EFI Stub: Exiting boot service...
`
if add 'devicetree /sc8180x-samsung-galaxy-book-s.dtb'  to grub.cfg (this dtb file from your 'linux-image-5.19.0-next-20220810-custom-..deb' file.), only show:
 `_`

Using ISO default kernel(5.11): 
Picture1. Do not plug in 2 typec hub(Failed and hang)
![1](https://github.com/aarch64-laptops/debian-cdimage/assets/10290601/bb892b52-b392-4bfe-a8b5-c60ee5c9a138)

Picture2. Plug in 2 typec hub(Success startup)
![2](https://github.com/aarch64-laptops/debian-cdimage/assets/10290601/0dd313b4-f13a-43e2-b6c9-9658bc66d549)

**Comment by LeosDing** [comment]
> > > Is your DTB compiled from the 5.19 kernel? Is this DTB file placed in /boot and added as a devicetree in grub.cfg?
> > 
> > 
> > Exactly ! ( as I said in [#16 (comment)](https://github.com/aarch64-laptops/debian-cdimage/issues/16#issuecomment-1231898585) )
> 
> Following your steps, the 5.19 kernel you compiled will display the following boot time: 
> `EFI stub: Booting Linux Kernel... `
> `EFI Stub: Using DTB from configuration table`  
> `EFI Stub: Exiting boot service...` 
>if add 'devicetree /sc8180x-samsung-galaxy-book-s.dtb' to grub.cfg (this dtb file from your 'linux-image-5.19.0-next-20220810-custom-..deb' file.), only show: 
> `_`

add 'console=tty0' under linux kernel line.

**Comment by LeosDing** [comment]
The 5.19 kernel you compiled requires the DTB file you provided to boot properly, and the DTB in this kernel cannot be used. The DTB in the 5.19 kernel that I compiled myself can boot the system normally.
WIFI is ok.
No longer requires two typec hubs to start.

Some issues:
1. 5.19 Kernel needs to add 'console=tty0'
2. Unable to recognize wired network card on typec hub
3. No sound card driver
4. Unable to identify battery and battery level
5. Touch screen cannot be used

Tomorrow, I will try the 6.0.6-battery kernel again and perhaps solve the above problems.

**Comment by oktaykarakiya** [comment]
Debian 12 came out, did anyone try?

**Comment by LeosDing** [comment]
> Debian 12 came out, did anyone try?

It's auto upgrade to Debian 12.

**Comment by oktaykarakiya** [comment]
So you had Debian 11 and now you have Debian 12? how is it? (aka did you try Debian 12)
Does everything work?

**Comment by LeosDing** [comment]
> So you had Debian 11 and now you have Debian 12? how is it? (aka did you try Debian 12) Does everything work?

For me, just like 11, there's no big difference.

**Comment by levitskyyy** [comment]
Linux still doesn't work on this laptop?

**Comment by Lirox-python** [comment]
Hi,
first of all, thank you very much for your great work – it’s really impressive that Linux runs on this device at all!

I just wanted to ask if there’s any way to get sound and Bluetooth working. I understand that this project is no longer very active, but I’m still very interested in making the laptop as fully functional as possible.

I also wanted to ask:
Would it be possible to try updating the older jenneron kernel 5.19.0 to a newer version, maybe just to see if it improves stability and performance?

Apart from that, everything works very well with Wayland – I’m currently using KDE Plasma.

Thanks again!

**Comment by pierro78** [comment]
the performance and stability were already good with Jenneron's kernel, I think the remaining issues were more like : 
. not drawing power from charger (?)
. no sound
. touchscreen not working

I don't know if there has been improvements on these issues ?

**Comment by Lirox-python** [comment]
Yeah, makes sense.

For me, sound, touchscreen, charging, brightness control, and Bluetooth still don't work.

I compiled a newer kernel today just to give it a try, and it felt faster for me.

I used the same .config and DTB from Jenneron's 5.19.0 kernel, but I still didn't have any keyboard or touchpad input – not even USB devices worked.

Do you maybe know how Jenneron managed to get input working back then? Maybe there was a specific patch or something else needed?

The kernel has improved a lot in recent years, so I'll definitely keep testing to see what's possible.
